### ISSUE TITLE
Option to leave group metadata unchanged on update

### ISSUE THREAD

**Issue by deewhyweb** [redhat-cop/group-sync-operator#39]
I've been testing the group-sync-operator in a PoC with Azure AD integration.  
The requirement I've been working to requires the groups created by the group-sync-operator to be augmented with metadata e.g. labels.  Once these groups are augmented we are then using the namespace-configuration-operator create namespaces and RBAC based on a GroupConfig object.

The problem is, when the group-sync-operator runs again (I have this running on a cron schedule) it removes any metadata added.
I've found a workaround which is to change the "group-sync-operator.redhat-cop.io/sync-provider" label, which causes the group-sync-operator to ignore this group when updating.  Problem with this though is any user additions / subtractions from this group are also ignored.

So I'm proposing a change to then group-sync-operator to leave the group untouched if it already exists, and just update users associated with the group.

**Comment by deewhyweb** [comment]
A proposed solution would be to be more explicit when setting the annotations and labels on the group.

Replacing 
```
if group.GetAnnotations() != nil {
	ocpGroupAnnotations = group.GetAnnotations()
}

if group.GetLabels() != nil {
	ocpGroupLabels = group.GetLabels()
}
ocpGroup.SetLabels(ocpGroupLabels)
ocpGroup.SetAnnotations(ocpGroupAnnotations)
```

With pseudo code (I'm not a Go developer)

```
if group.GetAnnotations() != nil {
	ocpGroupAnnotations = group.GetAnnotations()
}

if group.GetLabels() != nil {
	ocpGroupLabels = group.GetLabels()
}
for _, ocpGroupAnnotation := range ocpGroupAnnotations {
   ocpGroup.Annotations[ocpGroupAnnotation.key] = ocpGroupAnnotation.value
}

for _, ocpGroupLabel := range ocpGroupLabels {
   ocpGroup.Labels[ocpGroupLabel.key] = ocpGroupLabel.value
}
```

**Comment by sabre1041** [comment]
Resolved in #45
